### ISSUE TITLE
Unable to run in production and appliance modes locally

### ISSUE THREAD

**Issue by jvendetti** [ncbo/bioportal_web_ui#212]
Sometimes I want to run the Rails application in production or appliance modes locally for testing purposes. This used to work well, but lately I'm unable to do so due to `jQuery is not defined` errors:

```
Uncaught ReferenceError: jQuery is not defined  at jquery.validate.min.js:2:20436
Uncaught ReferenceError: jQuery is not defined at (index):33:5
```

**Reproduce for production mode:**

* Precompile the assets: `RAILS_ENV=production rails assets:precompile`
* Launch the Rails application: `RAILS_SERVE_STATIC_FILES=true rails s -e production`

This results in the `jQuery is not defined` error messages listed above (visible in the browser developer console). 

I've verified that `config/environments/production.rb` has the proper setting for serving assets:

`config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?`

**Reproduce for appliance mode:**

* Precompile the assets: `RAILS_ENV=appliance rails assets:precompile`
* Launch the Rails application: `RAILS_SERVE_STATIC_FILES=true rails s -e appliance`

This results in the identical set of `jQuery is not defined` error messages listed above.

I have no such issues running the application in development mode. I've spent half a day trying to track this down with no success.

**Comment by syphax-bouazzouni** [comment]
I tried to reproduce your error with our branch, and yes same error.

By checking the used assets in the browser I found that they were empty (which explains Jquery is not found), but the assets were correctly generated assets:precompile (checked by seeing the files in the public/assets folder).

So I found this same error: https://stackoverflow.com/questions/49758025/rails-5-1-puma-serving-empty-assets-in-production-mode
And the solution was to change in the appliance.rb and production.rb, the line `config.action_dispatch.x_sendfile_header = 'X-Sendfile'` to `config.action_dispatch.x_sendfile_header = 'X-Sendfile'  unless config.public_file_server.enabled`

I hope this was helpful.
